### ISSUE TITLE
Crash while running "terraform apply"

### ISSUE THREAD

**Issue by bnordbo** [hashicorp/terraform#9864]
This crash happens with Terraform 0.7.8. Reruns does not exhibit the same behaviour.

The config is rather big, so I cannot include it here. The crash happens when applying a fairly big diff on AWS infrastructure. The crash log is [here](https://gist.github.com/bnordbo/b86528d8ee408b39e0a8521abb627644).

This is the panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xf1f24]

goroutine 9336 [running]:
panic(0x25e3540, 0xc420010100)
        /usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
github.com/hashicorp/terraform/terraform.(*EvalApplyProvisioners).apply(0xc4207acea0, 0x433ec20, 0xc42069ba00, 0x0, 0x0)
        /private/tmp/terraform-20161101-67119-1rcq55o/terraform-0.7.8/src/github.com/hashicorp/terraform/terraform/eval_apply.go:278 +0x5b4
github.com/hashicorp/terraform/terraform.(*EvalApplyProvisioners).Eval(0xc4207acea0, 0x433ec20, 0xc42069ba00, 0x2, 0x2, 0xc420f23770, 0x24)
        /private/tmp/terraform-20161101-67119-1rcq55o/terraform-0.7.8/src/github.com/hashicorp/terraform/terraform/eval_apply.go:179 +0x151
github.com/hashicorp/terraform/terraform.EvalRaw(0x4324480, 0xc4207acea0, 0x433ec20, 0xc42069ba00, 0x29882a0, 0x0, 0x0, 0x0)
        /private/tmp/terraform-20161101-67119-1rcq55o/terraform-0.7.8/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x17f
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc42039cb60, 0x433ec20, 0xc42069ba00, 0x2, 0x2, 0xc42058aab0, 0x24)
        /private/tmp/terraform-20161101-67119-1rcq55o/terraform-0.7.8/src/github.com/hashicorp/terraform/terraform/eval_sequence.go:10 +0x6c
github.com/hashicorp/terraform/terraform.EvalRaw(0x4324d80, 0xc42039cb60, 0x433ec20, 0xc42069ba00, 0x4324d80, 0xc42039cb60, 0xc4215b46e0, 0x4a)
        /private/tmp/terraform-20161101-67119-1rcq55o/terraform-0.7.8/src/github.com/hashicorp/terraform/terraform/eval.go:53 +0x17f
github.com/hashicorp/terraform/terraform.Eval(0x4324d80, 0xc42039cb60, 0x433ec20, 0xc42069ba00, 0xc42039cb60, 0x4324d80, 0xc42039cb60, 0xc41ffdbe58)
        /private/tmp/terraform-20161101-67119-1rcq55o/terraform-0.7.8/src/github.com/hashicorp/terraform/terraform/eval.go:34 +0x53
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x2a939e0, 0xc42095e1b0, 0x0, 0x0)
        /private/tmp/terraform-20161101-67119-1rcq55o/terraform-0.7.8/src/github.com/hashicorp/terraform/terraform/graph.go:249 +0xa0f
github.com/hashicorp/terraform/dag.(*AcyclicGraph).Walk.func3(0xc423902050, 0xc421a94000, 0xc423902090, 0xc421e17110, 0xc4239020a0, 0x2a939e0, 0xc42095e1b0, 0xc423179a40, 0xc421343860)
        /private/tmp/terraform-20161101-67119-1rcq55o/terraform-0.7.8/src/github.com/hashicorp/terraform/dag/dag.go:238 +0x251
created by github.com/hashicorp/terraform/dag.(*AcyclicGraph).Walk
        /private/tmp/terraform-20161101-67119-1rcq55o/terraform-0.7.8/src/github.com/hashicorp/terraform/dag/dag.go:247 +0x618
```

**Comment by philipsd6** [comment]
Dunno if that :+1: makes sense &mdash; I mean I'm having this issue too. Tried with and without the `-Xnew-apply` flag.  However, re-running the apply successfully finished it up.

**Comment by mitchellh** [comment]
This is fixed in 0.7.9. Note that the apply did complete successfully this was a post-apply crash.
